jib-core/src/main/java/com/google/cloud/tools/jib/api/ImageReference.java

```diff
@@ -360,11 +405,12 @@ public boolean equals(Object other) {
     ImageReference otherImageReference = (ImageReference) other;
     return registry.equals(otherImageReference.registry)
         && repository.equals(otherImageReference.repository)
-        && tag.equals(otherImageReference.tag);
+        && tag.equals(otherImageReference.tag)
+        && Objects.equals(digest, otherImageReference.digest);
```

Comment:
I know `digest` is of `Optional`, but is this different from `digest.equals(otherImageReference.digest)`?

Reply:
digest is a `@Nullable` string, so `Objects.equals` is used here to avoid NPE